src/main/java/com/google/api/codegen/transformer/php/PhpGapicSurfaceTestTransformer.java

```diff
@@ -73,11 +76,7 @@
   private final PhpFeatureConfig featureConfig = new PhpFeatureConfig();
   private final MockServiceTransformer mockServiceTransformer = new MockServiceTransformer();
 
-  private final PackageMetadataConfig packageConfig;
-
-  public PhpGapicSurfaceTestTransformer(PackageMetadataConfig packageConfig) {
-    this.packageConfig = packageConfig;
-  }
+  public PhpGapicSurfaceTestTransformer() {}
```

Comment:
Is it necessary to remove `PackageMetadataConfig` from the constructor of `PhpGapicSamplesTransformer` as well?

Reply:
@hzyi-google Yes, this is the topic I wanted to discuss with @vchudnov-g  today. It does not have to be done in this PR, as it is a bigger change. 

The idea is that samplegen does not rely on `PackageMetadataConfig` for anything in any language. That object is created from a `.yaml` file, which is passed as `--package_yaml2` argument to the generator. That argument is a legacy thing (Garrett started removing dependency on it long time ago, removing most of the stuff from it and renaming it from `--package_yaml` to `--package_yaml2`). The problem with it is that it is generated by artman from artman yaml and it contains only packaging-related stuff (actually it contains almost no useful information at this point).

Since artman will go away, the artman_yaml will go too.

Reply:
That sounds good to me. IIRC it should be very easy to decouple `PackageMetadataConfig` and `LangGapicSamplesTransformer`, at least in most languages.